### PR TITLE
Issue/270 thought bubble

### PR DESCRIPTION
**Pull request by phfatmonkey**
Thought Bubble System for characters and computers

To test, open the [Test] Character Thought Bubble scene. All the character prefabs are there. Each one has a nested speech bubble prefab. Edit the text and set to active to see how it looks in game.

image

The properties circled in red are what you would need to update at runtime:

**Active**: turns the thought bubble on and off
**Starting Thought Bubble Text**: Currently this is updating every frame so we can test while the game is running

The properties in green would be set when this prefab is added to the character prefabs

**Center of Mass**: The hip joint. Thought bubble is placed above his head relative to this joint, so it doesn't stay up in the air when the character sits down.
**Vertical Offset**: The height above the hip joint where the thought bubble is placed. Mostly the same, but can be fine-tuned for each character

### CODE REVIEW

File: Assets/Code/Character/CameraFacing.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  [SerializeField] private Transform centerOfMass;
+  public float verticalOffset = 1f;
+  private void Start () {
+    mainCamera = Camera.main;
+  }
+  private void Update () {
+    Vector3 vector = mainCamera.transform.eulerAngles;
+    // Vector3 vector = new Vector3(mainCamera.transform.position.x - this.transform.position.x, mainCamera.transform.position.y - this.transform.position.y, mainCamera.transform.position.z - this.transform.position.z);
+    // vector.Normalize();
+    this.transform.eulerAngles = vector;
```
Comment (CowboyCoderForever) [inline Assets/Code/Character/CameraFacing.cs]: I think you can do something as easy as:
`this.transform.LookAt(mainCamera.transform);`

File: Assets/Code/Character/ThoughtBubble.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+{
+  
+  [SerializeField] private bool _Active;
+  [SerializeField] private string _startingThoughtBubbleText;
+  [SerializeField] private GameObject _thoughtBubbleRoot;
+
+  [SerializeField] private TMP_Text _thoughtBubbleTextComponent;
+  [SerializeField] private Image _thoughtBubbleSprite;
+  private string _thoughtBubbleText;
+  private bool _active;
```
Comment (CowboyCoderForever) [inline Assets/Code/Character/ThoughtBubble.cs]: I think you have an extra "active" boolean here. You can just use the `_Active` as your c# property "backing field", as they call it.

File: Assets/Code/Character/CameraFacing.cs (hunk 1 above)
Comment (phfatmonkey) [reply Assets/Code/Character/CameraFacing.cs]: Fixed. Thanks, this is much simpler!

File: Assets/Code/Character/ThoughtBubble.cs (hunk 2 above)
Comment (phfatmonkey) [reply Assets/Code/Character/ThoughtBubble.cs]: Fixed. This was only for debug purposes anyway.

**Comment by phfatmonkey** [comment]
@CowboyCoderForever this should be ready now

**Pull request merged**